### PR TITLE
Edit the storage to include storage of isMarked

### PR DESCRIPTION
**Pull request by jasperng-nus**
Change the JsonAdaptedCompany class to include the isMarked variable in the storage so that whenever the user mark or unmark a company it will be stored in the storage

Changed the TypicalCompany and UnsortedTypicalCompany to match with our context as companies instead of persons, made some changes to make our code fit our context better

### CODE REVIEW

File: src/main/java/seedu/address/storage/JsonSerializableInternBook.java [hunk 1]
```diff
@@ -14,20 +14,20 @@
...
 
 /**
- * An Immutable AddressBook that is serializable to JSON format.
+ * An Immutable InternBook that is serializable to JSON format.
  */
 @JsonRootName(value = "addressbook")
-class JsonSerializableAddressBook {
+class JsonSerializableInternBook {
 
     public static final String MESSAGE_DUPLICATE_PERSON = "Persons list contains duplicate company(s).";
```
Comment (SherwynNg) [inline src/main/java/seedu/address/storage/JsonSerializableInternBook.java:22]: This should be MESSAGE_DUPLICATE_COMPANY and the message should be changed

File: src/main/java/seedu/address/storage/JsonSerializableInternBook.java [hunk 2]
```diff
@@ -14,20 +14,20 @@
...
 
-    private final List<JsonAdaptedPerson> persons = new ArrayList<>();
+    private final List<JsonAdaptedCompany> persons = new ArrayList<>();
 
     /**
      * Constructs a {@code JsonSerializableAddressBook} with the given persons.
      */
     @JsonCreator
-    public JsonSerializableAddressBook(@JsonProperty("persons") List<JsonAdaptedPerson> persons) {
+    public JsonSerializableInternBook(@JsonProperty("persons") List<JsonAdaptedCompany> persons) {
```
Comment (SherwynNg) [inline src/main/java/seedu/address/storage/JsonSerializableInternBook.java:30]: should we change this to companies

File: src/main/java/seedu/address/storage/JsonSerializableInternBook.java [hunk 3]
```diff
@@ -36,8 +36,8 @@ public JsonSerializableAddressBook(@JsonProperty("persons") List<JsonAdaptedPers
      *
      * @param source future changes to this will not affect the created {@code JsonSerializableAddressBook}.
      */
-    public JsonSerializableAddressBook(ReadOnlyInternBook source) {
-        persons.addAll(source.getCompanyList().stream().map(JsonAdaptedPerson::new).collect(Collectors.toList()));
+    public JsonSerializableInternBook(ReadOnlyInternBook source) {
+        persons.addAll(source.getCompanyList().stream().map(JsonAdaptedCompany::new).collect(Collectors.toList()));
```
Comment (SherwynNg) [inline src/main/java/seedu/address/storage/JsonSerializableInternBook.java:40]: this should be companies.addAll instead

File: src/main/java/seedu/address/storage/JsonSerializableInternBook.java [hunk 4]
```diff
@@ -47,7 +47,7 @@ public JsonSerializableAddressBook(ReadOnlyInternBook source) {
      */
     public InternBook toModelType() throws IllegalValueException {
         InternBook internBook = new InternBook();
-        for (JsonAdaptedPerson jsonAdaptedPerson : persons) {
+        for (JsonAdaptedCompany jsonAdaptedPerson : persons) {
```
Comment (SherwynNg) [inline src/main/java/seedu/address/storage/JsonSerializableInternBook.java:50]: should change these jsonAdaptedPerson to jsonAdaptedCompany instead

File: src/main/java/seedu/address/storage/JsonSerializableInternBook.java [hunk 5]
```diff
@@ -47,7 +47,7 @@ public JsonSerializableAddressBook(ReadOnlyInternBook source) {
...
             Company company = jsonAdaptedPerson.toModelType();
```
Comment (SherwynNg) [inline src/main/java/seedu/address/storage/JsonSerializableInternBook.java:51]: this one as well

**Comment by Xilef121** [comment]
LGTM

**Comment by AjayArvind2207** [comment]
LGTM on the whole 👍

**Pull request merged**